workers/generic-worker/os_groups_insecure.go

```diff
@@ -13,11 +15,57 @@ type OSGroups struct {
 }
 
 func (osGroups *OSGroups) Start() *CommandExecutionError {
-	if len(osGroups.Task.Payload.OSGroups) > 0 {
-		return MalformedPayloadError(fmt.Errorf("osGroups feature is not supported on platform %v - please modify task definition and try again", runtime.GOOS))
+	groupNames := osGroups.Task.Payload.OSGroups
+	if len(groupNames) == 0 {
+		return nil
 	}
+
+	userGroups, err := currentUserGroups()
+	if err != nil {
+		return executionError(internalError, errored, fmt.Errorf("failed to get user groups: %v", err))
+	}
+
+	notInGroups := checkUserGroupsInList(groupNames, userGroups)
+	if len(notInGroups) > 0 {
+		return MalformedPayloadError(fmt.Errorf("current user is a member of groups: %v, but is not a member of: %v", userGroups, notInGroups))
```

Comment:
```suggestion
		return MalformedPayloadError(fmt.Errorf("Task definition contains unsupported osGroups: %v. Allowed values (on this worker pool): %v", notInGroups, userGroups))
```
Does the list include the primary group (e.g. task_XXXXXXXXX)? If so, is there a way to filter that one out, just because that will be variable and is unpredictable, so you can't actually use it. No worries if it isn't trivial.

Reply:
Yes this includes the primary group too. I can filter this out no problem. Thanks for flagging.